File: RoRClient/Commands/Game/Other/ChangeMapSelectionCommand.cs
using RoRClient.Commands.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RoRClient.Communication.DataTransferObject;
using RoRClient.Models.Session;

namespace RoRClient.Commands.Game.Other
{
	class ChangeMapSelectionCommand : ChangeMapSelectionCommandBase
	{
		public ChangeMapSelectionCommand(RoRSession session, MessageInformation message) : base(session, message)
		{
		}
	}
}

File: RoRServer/src/main/java/commands/game/ChangeMapSelectionCommand.java
package commands.game;

import commands.base.CommandBase;
import communication.MessageInformation;
import models.session.GameSession;
import models.session.RoRSession;

/**
 * Setzt den neuen MapName der GameSession
 */
public class ChangeMapSelectionCommand extends CommandBase{

	private String mapName;
	
	public ChangeMapSelectionCommand(RoRSession session, MessageInformation messageInfo) {
		super(session, messageInfo);
		this.mapName = messageInfo.getValueAsString("mapName");
	}

	@Override
	public void execute() {
		GameSession gameSession = (GameSession)session;
		gameSession.setMapName(mapName);
	}

}
